gcc/cp/contracts.cc

```diff
@@ -2632,7 +2658,26 @@ apply_postconditions (tree fndecl)
   if (outline_contracts_p (fndecl))
     add_post_condition_fn_call (fndecl);
   else
-    emit_postconditions (DECL_CONTRACTS (fndecl));
+    {
+      /* We only care about `noexcept()` case. Functions with `throw`
+	 exception specification may not work in presence of throwing
```

Comment:
How are functions with 'throw' any different from 'noexcept' functions in that regard? Not all that big a deal, the former have been deprecated and removed.

Reply:
Suggestion: "We only care about the noexcept/noexcept(true)/throw() cases. Functions with e.g. throw(int) exception specifications are not covered here, and being both deprecated and removed, are not relevant for this functionality."

Reply:
that seems reasonable, since there can be no existing code with a violation handler, there's no need to support it in 'gnu++26' mode (I did not yet check if we intend to support `thow(blah)` in gnu++26 in general).

Reply:
GCC rejects void f() throw(blah) in c++17 and gnu++17 mode and above.

Reply:
great  - so we do not need to worry about EH_SPEC_BLOCK tree - which is what you get for a dynamic exception spec

Reply:
"for non-outlined contract checks, make the contract check noexcept for noexcept functions"